From 9c165a4f87ea3914428f64f8262e42eea3c86877 Mon Sep 17 00:00:00 2001
From: David Ortiz <daveortiz@gmail.com>
Date: Mon, 6 Dec 2021 19:22:17 -0500
Subject: Specify a default metrics format (#36371)

This fixes an issue where a blank Content-Type header would
be returned in cases where application metrics were disabled
as would be the case pilot-agent is running as a gateway.

This addresses https://github.com/istio/istio/issues/3633
---
 pilot/cmd/pilot-agent/status/server.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/pilot/cmd/pilot-agent/status/server.go b/pilot/cmd/pilot-agent/status/server.go
index cd65c0434a..a90aec51b0 100644
--- a/pilot/cmd/pilot-agent/status/server.go
+++ b/pilot/cmd/pilot-agent/status/server.go
@@ -467,6 +467,9 @@ func (s *Server) handleStats(w http.ResponseWriter, r *http.Request) {
 			metrics.AppScrapeErrors.Increment()
 		}
 		format = negotiateMetricsFormat(contentType)
+	} else {
+		// Without app metrics format use a default
+		format = expfmt.FmtText
 	}
 
 	if agent, err = scrapeAgentMetrics(); err != nil {
-- 
2.35.3

